### PR TITLE
enable dot in attribute when user profile enabled

### LINKED ISSUE #24918: User details tab does not display or update attibutes with dot
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

When a user profile attribute contains a dot like "with.dot", the user details tab fails to display and update the value of this attribute.

Stored values are not displayed:
image

Update does not work:
screen-record-dot-not-stored

### Version

22.0.5 and nightly

### Expected behavior

Attributes which have a dot in the key should work like any others

### Actual behavior

Attributes not displayed. The PUT update request sends the following, when value "Foo" was entered to be stored:

```
	"attributes": {
		"with": {
			"dot": "Foo"
		},
		"with.dot": [
			"Baz"
		],
		"without_dot": [
			"Bar"
		]
	}
    ...
```

### How to Reproduce?

* Enable user profile
* Declare an attribute with dot .
* Try to view or update the value

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by edewit**
fixes: #24918
Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

### CODE REVIEW

File: js/libs/ui-shared/src/user-profile/utils.ts [hunk 1]
```diff
@@ -44,9 +44,10 @@ export const isRootAttribute = (attr?: string) =>
   attr && ROOT_ATTRIBUTES.includes(attr);
 
 export const fieldName = (name?: string) =>
-  `${
-    isRootAttribute(name) ? "" : "attributes."
-  }${name}` as FieldPath<UserFormFields>;
+  `${isRootAttribute(name) ? "" : "attributes."}${name?.replaceAll(
```
Comment (jonkoops) [inline js/libs/ui-shared/src/user-profile/utils.ts:47]: Can't we call `beerify` here?

**Review by jonkoops — APPROVED** [review]

**Review by jonkoops — DISMISSED** [review]

**Review by jonkoops — APPROVED** [review]

**Comment by sschu** [comment]
@ssilvert Can we also back-port this to KC23? It is currently a blocker for us for KC23...

**Comment by jonkoops** [comment]
@sschu marked this for backporting, but the patches do not apply cleanly unfortunately, so this might be a little more of a hassle to backport.

**Comment by sschu** [comment]
Thanks @jonkoops ! Did we already divert from KC23 that much?

**Pull request merged**